test/jdk/java/lang/invoke/indify/Indify.java

```diff
@@ -356,118 +448,206 @@ protected Class<?> findClass(String name) throws ClassNotFoundException {
+                            methodEntry = (MethodRefEntry) newClassModel.constantPool().entryByIndex(ref2i);
+                        }
+
+                        if(methodEntry == null || !"invokeExact".equals(methodEntry.nameAndType().name().stringValue())){
+                            System.err.println(m+": Failed to create invokedynamic at "+i.opcode().bytecode());
                             continue;
                         }
-                        String invType = pool.getString(ref2[2]);
+
+                        String invType = methodEntry.type().stringValue();
                         String bsmType = indySignatures.get(conm);
                         if (!invType.equals(bsmType)) {
-                            System.err.println(m+": warning: "+conm+" call type and local invoke type differ: "
-                                    +bsmType+", "+invType);
+                            System.err.println(m+": warning: "+conm+" call type and local invoke type differ: " + bsmType+", " + invType);
                         }
-                        assert(i.len == 3 || i2.len == 3);
-                        if (!quiet)  System.err.println(i+" "+conm+";...; "+i2+" => invokedynamic "+con);
-                        int start = i.pc + 3, end = i2.pc;
-                        System.arraycopy(i.codeBase, start, i.codeBase, i.pc, end-start);
-                        i.forceNext(0);  // force revisit of new instruction
-                        i2.u1AtPut(-3, opc_invokedynamic);
-                        i2.u2AtPut(-2, con.index);
-                        i2.u2AtPut(0, (short)0);
-                        i2.u1AtPut(2, opc_nop);
-                        //System.out.println(new Instruction(i.codeBase, i2.pc-3));
+
+                        assert (i.sizeInBytes() == 3 || i2.sizeInBytes() == 3);
+                        System.err.println("----------------------------------------------------------------Transforming Method INDY Instructions & Creating New ClassModels------------------------------------------------------------------}}}");
+                        if (!quiet) System.err.println(":::Transfmoring the Method: "+ m.methodName() +" instruction: " + i + " invokedynamic: " + con.index() );
+                        MethodModel finalConm = conm;
+                        codeTransform = (b, e) ->{
+                            String a1 = null, a2 = null;
+                            if(e instanceof InvokeInstruction){
+                                a1 = ((InvokeInstruction) e).method().name().stringValue();
+                                a2 = finalConm.methodName().stringValue();
+                            }
+                            if (e instanceof InvokeInstruction && Objects.equals(a1, a2)) {
+                                System.err.println(">> Removing instruction invokestatic for Method: " + ((InvokeInstruction) e).name());
+                                b.andThen(b);
```

Comment:
This will be removed.
I was forcing the classBuilder to chain itself effectively appending no new elements and removing the current e "CodeElement I,e the instruction".